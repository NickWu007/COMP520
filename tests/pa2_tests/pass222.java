// PA2 pass AST references in arguments and method invocations
class C {
    public void foo() {
        // "this" as reference and as qualifier
        this.foo(3, this);
        other.foo(4, other);
    }
}

