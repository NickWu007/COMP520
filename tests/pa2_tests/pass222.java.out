======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "C" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "foo" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [2]
  .   .   . CallStmt
  .   .   .   QRef
  .   .   .     "foo" Identifier
  .   .   .     ThisRef
  .   .   .   ExprList [2]
  .   .   .   . LiteralExpr
  .   .   .   .   "3" IntLiteral
  .   .   .   . RefExpr
  .   .   .   .   ThisRef
  .   .   . CallStmt
  .   .   .   QRef
  .   .   .     "foo" Identifier
  .   .   .     IdRef
  .   .   .       "other" Identifier
  .   .   .   ExprList [2]
  .   .   .   . LiteralExpr
  .   .   .   .   "4" IntLiteral
  .   .   .   . RefExpr
  .   .   .   .   IdRef
  .   .   .   .     "other" Identifier
=============================================
